code/modules/reagents/reagent_containers/glass.dm

```diff
@@ -198,7 +198,8 @@
 		filling.icon += mix_color_from_reagents(reagents.reagent_list)
 		filling.alpha = mix_alpha_from_reagents(reagents.reagent_list)
 
-		overlays += filling
+		if (!opaque)
```

Comment:
Just avoid creating the image in the first place if you're not going to use it

Reply:
I believe this was what was causing the issue that lead to lids being removed from the beakers. somehow the overlay of the bloodbag was appearing whenever the small bluespace beaker, small stasis and large stasis were called. #14353 to be exact. correct me if I'm wrong because I don't understand what I'm doing wrong.

Reply:
No, I'm sure the issue with lids is that the code just underneath this was not being executed.
Instead of checking `!opaque` here, move the check before the creation of the image on L178, like: `if(!opaque && reagents.total_volume)`